test/e2e/helpers.go

```
@@ -0,0 +1,609 @@
+		resp.Body.Close()
+		if err != nil {
+			return "", fmt.Errorf("failed to read response body: %w", err)
+		}
+
+		if resp.StatusCode == http.StatusOK {
+			return string(body), nil
+		}
+
+		// Retry on 404 (ingress not ready yet)
+		if resp.StatusCode == http.StatusNotFound {
+			lastErr = fmt.Errorf("unexpected status code: %d, body: %s", resp.StatusCode, strings.TrimSpace(string(body)))
```

golangci-lint (2.5.0)

[error] do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unexpected status code: %d, body: %s\", resp.StatusCode, strings.TrimSpace(string(body)))"

(err113)